lib/vtk/commands/socks/setup.rb

```diff
@@ -0,0 +1,381 @@
+          return true if installed
+
+          log 'Homebrew not installed. Installing now...'
+          system '/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"'
+        end
+
+        def ssh_config_clean_up
+          FileUtils.rm_rf '/tmp/dova-devops'
+        end
+
+        def backup_existing_ssh_config
+          return true unless ssh_config_exists?
+
+          if File.exist? "#{ssh_config_path}.bak"
+            log "!!! ERROR: Could not make backup of #{pretty_ssh_config_path} as #{pretty_ssh_config_path}.bak " \
+              'exists. Aborting.'
+            exit 1
+          end
+
+          FileUtils.mv ssh_config_path, "#{ssh_config_path}.bak"
+        end
+
+        def create_ssh_directory
+          ssh_dir = File.dirname ssh_config_path
+          FileUtils.mkdir_p ssh_dir
+          FileUtils.chmod 0o700, ssh_dir
+        end
+
+        def configure_ssh_config_with_keychain
+          return unless macos?
+          return if ssh_config_configured_with_keychain?
+
+          keychain_config = <<~CFG
+
+            # Maintain SSH keys in macOS Keychain
+            Host *
+              UseKeychain yes
+              AddKeysToAgent yes
+              IdentityFile #{pretty_ssh_key_path}
+          CFG
+
+          IO.write ssh_config_path, keychain_config, mode: 'a'
+        end
+
+        def ssh_config_configured_with_keychain?
+          return false unless ssh_config_exists?
+
+          ssh_config_local = File.readlines ssh_config_path
+          ssh_config_local.grep(/UseKeychain yes/).size.positive?
+        end
+
+        def ssh_agent_add
+          FileUtils.chmod 0o600, ssh_key_path if key_exists?
+          FileUtils.chmod 0o600, "#{ssh_key_path}.pub" if public_key_exists?
+
+          if macos?
+            `ssh-add -K 2> /dev/null; ssh-add -K #{ssh_key_path} 2> /dev/null`
+          elsif ubuntu?
+            `[ -z "$SSH_AUTH_SOCK" ] && eval "$(ssh-agent -s)";
+              ssh-add 2> /dev/null; ssh-add #{ssh_key_path} 2> /dev/null`
+          end
+        end
+
+        def test_ssh_connection
+          output.print '----> Testing SOCKS SSH connection...'
+
+          add_ip_to_known_hosts
+
+          ssh_output = `ssh -i #{ssh_key_path} -F #{ssh_config_path} -o ConnectTimeout=5 -q socks -D #{port} exit 2>&1`
+
+          if ssh_output.include? 'This account is currently not available.'
+            output.puts ' ✅'
+          else
+            check_ssh_error ssh_output
+            exit 1
+          end
+        end
+
+        def add_ip_to_known_hosts
+          jump_box_ip = `grep -A 2 'Host socks' ~/.ssh/config | grep ProxyCommand | awk '{print $6}'`.chomp
+          socks_ip = `grep -A 2 'Host socks' ~/.ssh/config | grep HostName | awk '{print $2}'`.chomp
+
+          `ssh-keygen -R #{jump_box_ip}` if File.exist? '~/.ssh/known_hosts'
+          `ssh-keygen -R #{socks_ip}` if File.exist? '~/.ssh/known_hosts'
```

Comment:
I don't think this line works.

I think it would need to look similar to line 230 below.